data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -354,15 +361,19 @@ private AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest> createBulkReq
             int index = 0;
             for (final BulkResponseItem bulkItemResponse : response.items()) {
                 BulkOperationWrapper bulkOperation =
-                    (BulkOperationWrapper)request.getOperationAt(index);
+                        (BulkOperationWrapper)request.getOperationAt(index);
                 if (isItemInError(bulkItemResponse)) {
                     if (existingDocumentQueryManager != null && POTENTIAL_DUPLICATES_ERRORS.contains(bulkItemResponse.status())) {
                         existingDocumentQueryManager.addBulkOperation(bulkOperation);
                         index++;
                         continue;
                     }
 
-                    if (!NON_RETRY_STATUS.contains(bulkItemResponse.status())) {
+                    boolean isGenerallyRetryable = !NON_RETRY_STATUS.contains(bulkItemResponse.status());
+                    boolean isDeleteNotFound = bulkItemResponse.status() == RestStatus.NOT_FOUND.getStatus() &&
+                            bulkItemResponse.operationType() == OperationType.Delete;
+
+                    if (isGenerallyRetryable || isDeleteNotFound) {
```

Comment:
Related to the question of max retries from the original issue (see [comment](https://github.com/opensearch-project/data-prepper/issues/5521#issuecomment-2812636129)):

What if the delete operation is for a document that never existed? This would keep retrying because there is not a create operation in the bulk request. What would it take to see if a create/index operation exists in the current request?